libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c

```diff
@@ -1131,7 +1131,23 @@ static OTA_Err_t prvProcessDataHandler( OTA_EventData_t * pxEventData )
 
             if( xErr != kOTA_Err_None )
             {
-                OTA_LOG_L2( "[%s] Failed to update job status %d\r\n", OTA_METHOD_NAME, xErr );
+                OTA_LOG_L1( "[%s] Failed to update job status %d\r\n", OTA_METHOD_NAME, xErr );
+
+                /* Job status update failed, call the platform specific code to abort the image. */
+                xErr = xOTA_Agent.xPALCallbacks.xSetPlatformImageState( xOTA_Agent.ulServerFileID, eOTA_ImageState_Aborted );
+
+                if( xErr != kOTA_Err_None )
+                {
+                    OTA_LOG_L1( "[%s] Error trying to set platform image state (0x%08x)\r\n", OTA_METHOD_NAME, ( int32_t ) xErr );
```

Comment:
should you call the complete callback on line 1145 with a different error code if the PAL has failed here?